src/others/xml/parsers/lv_xml_buttonmatrix_parser.c

```
@@ -0,0 +1,170 @@
+/**
+ * @file lv_xml_buttonmatrix_parser.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_xml_buttonmatrix_parser.h"
+#if LV_USE_XML
+
+#include "../../../lvgl.h"
+#include "../../../lvgl_private.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+static lv_buttonmatrix_ctrl_t ctrl_text_to_enum_value(const char * txt);
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+void * lv_xml_buttonmatrix_create(lv_xml_parser_state_t * state, const char ** attrs)
+{
+    LV_UNUSED(attrs);
+
+    void * item = lv_buttonmatrix_create(lv_xml_state_get_parent(state));
+    return item;
+}
+
+void lv_xml_buttonmatrix_apply(lv_xml_parser_state_t * state, const char ** attrs)
+{
+    void * item = lv_xml_state_get_item(state);
```

Okay I should not be too strict but, function that does not return error we should provide parameter validation plus a LOG_ERR for tracking, otherwise silent failures would be a pain to debug.

---

Now the assumption is that the XML is always correct as it's coming from the editor which has a validator. 

Regardless of it,  agree that basic asserts would be useful. These are missing from everywhere not only from the button matrix, so I'd add them in a new PR.

---

Fair enough, I will be resolving since it is a larger issue by itself the miss of some asserts